packages/edit-site/src/components/global-styles/block-preview-panel.js

```diff
@@ -4,25 +4,30 @@
 import { BlockPreview } from '@wordpress/block-editor';
 import { getBlockType, getBlockFromExample } from '@wordpress/blocks';
 import { useResizeObserver } from '@wordpress/compose';
+import { __experimentalSpacer as Spacer } from '@wordpress/components';
 
 const BlockPreviewPanel = ( { name } ) => {
-	const blockExample = getBlockType( name )?.example;
 	const [
 		containerResizeListener,
 		{ width: containerWidth, height: containerHeight },
 	] = useResizeObserver();
+	const blockExample = getBlockType( name )?.example;
+	const blocks = blockExample && getBlockFromExample( name, blockExample );
 	const viewportWidth = blockExample?.viewportWidth || containerWidth;
+	const minHeight = containerHeight;
 
 	return ! blockExample ? null : (
-		<div className="edit-site-global-styles__block-preview-panel">
-			{ containerResizeListener }
+		<Spacer marginX={ 4 } marginBottom={ 4 }>
+			<div className="edit-site-global-styles__block-preview-panel">
+				{ containerResizeListener }
 
-			<BlockPreview
-				viewportWidth={ viewportWidth }
-				__experimentalMinHeight={ containerHeight }
-				blocks={ getBlockFromExample( name, blockExample ) }
-			/>
-		</div>
+				<BlockPreview
```

Comment:
BlockPreview uses `memo`, I was hoping the preview to not re-render on forward and back navigation.
However, it still renders. 
Also tried wrapping BlockPreviewPanel component with `memo`, yet it re-renders. 
Please suggest.

Reply:
Hmm.. is it re-rendering because the `clientId` of the elements is changing? I guess `memo` should stop this happening, but it still re-renders?

Reply:
You can't avoid re-renders unless you're using the same component instance. So you'd need the same `<BlockPreview />` call for all the previews.